pkg/controller/machine/machine_controller.go

```diff
@@ -310,3 +324,93 @@ func (r *ReconcileMachine) getMachinesInCluster(ctx context.Context, namespace,
 
 	return machines, nil
 }
+
+
+// Writer implements io.Writer interface as a pass-through for klog.
+type Writer struct {
```

Comment:
If this merges, we can remove this type and it's method: https://github.com/kubernetes/klog/pull/76

Reply:
Can we remove the need for this writer and just use `klog` everywhere in side the library instead of passing it in?

Reply:
@vincepri the idea here is copying the stuff from kubectl/drain with as few changes as possible.  If we make a bunch of changes, it will be harder to maintain in the future for us.  We should just consume whatever the public interfaces are.  I don't think passing in a writer is that big of a deal.

Reply:
In that case, I'd prefer to add a default in the library instead of having a `Writer` type in the Machine Controller package, which seems really out of place

Reply:
I agree it seems out of place here.  I opened a PR against klog so it could live there.  But if we want to capture additional information for logging (as you suggested below), we might want to own that function.  Feel free to chime in on my klog PR if you think it should live there: https://github.com/kubernetes/klog/pull/76

Maybe it should live in client-go.  Either way, we need to do *something* and this is a good stand in for now IMO.

Reply:
If the logger writer has to remain, and the library can’t be changed, please move this writer to the util package, called something like LogWriter with a comment that it should be removed down the line

Reply:
I can move it to util.